src/transformers/models/clip/modeling_tf_clip.py

```diff
@@ -0,0 +1,1528 @@
+            return_dict=inputs["return_dict"],
+            training=inputs["training"],
+        )
+
+        return text_model_outputs
+
+
+class TFCLIPVisionTransformer(tf.keras.layers.Layer):
+    def __init__(self, config: CLIPVisionConfig, **kwargs):
+        super().__init__(**kwargs)
+
+        self.embeddings = TFCLIPVisionEmbeddings(config, name="embeddings")
+        self.pre_layrnorm = tf.keras.layers.LayerNormalization(epsilon=config.layer_norm_eps, name="pre_layrnorm")
+        self.encoder = TFCLIPEncoder(config, name="encoder")
+        self.post_layernorm = tf.keras.layers.LayerNormalization(epsilon=config.layer_norm_eps, name="post_layernorm")
+
+    def call(
+        self,
+        pixel_values: TFModelInputType,
+        output_attentions: bool,
+        output_hidden_states: bool,
+        return_dict: bool,
+        training: bool = False,
+        **kwargs,
+    ) -> Union[TFBaseModelOutputWithPooling, Tuple[tf.Tensor]]:
+
+        embedding_output = self.embeddings(pixel_values=pixel_values)
+        embedding_output = self.pre_layrnorm(inputs=embedding_output)
+
+        encoder_outputs = self.encoder(
+            hidden_states=embedding_output,
+            attention_mask=None,
+            causal_attention_mask=None,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            training=training,
+        )
+
+        sequence_output = encoder_outputs[0]
+        pooled_output = sequence_output[:, 0, :]
+        pooled_output = self.post_layernorm(inputs=pooled_output)
+
+        if not return_dict:
+            return (
+                sequence_output,
+                pooled_output,
+            ) + encoder_outputs[1:]
+
+        return TFBaseModelOutputWithPooling(
+            last_hidden_state=sequence_output,
+            pooler_output=pooled_output,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
+
+
+@keras_serializable
+class TFCLIPVisionMainLayer(tf.keras.layers.Layer):
+    config_class = CLIPVisionConfig
+
+    def __init__(self, config: CLIPVisionConfig, **kwargs):
+        super().__init__(**kwargs)
+
+        self.config = config
+
+        self.vision_model = TFCLIPVisionTransformer(config, name="vision_model")
+
+    def get_input_embeddings(self) -> tf.keras.layers.Layer:
+        return self.vision_model.embeddings
+
+    def call(
+        self,
+        pixel_values: Optional[TFModelInputType] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        training: bool = False,
+        **kwargs,
+    ) -> Union[TFBaseModelOutputWithPooling, Tuple[tf.Tensor]]:
+        inputs = input_processing(
+            func=self.call,
+            config=self.config,
+            input_ids=pixel_values,
```

Comment:
I'm not familiar with this. But this seems nit hackish and as there are going to be more vision models IMO it should be handled explicitly rather than setting `input_ids=pixel_values`. cc @Rocketknight1